content/post/performance-metrics-histogram.md

```diff
@@ -0,0 +1,245 @@
++++
+title = "Latency Histograms and Percentile Distributions In Wallaroo Performance Metrics"
+date = 2018-02-15T07:11:00-05:00
+draft = false
+author = "nisanharamati"
+description = "How we implemented Wallaroo's low overhead performance counters and the philosophy behind the choices we made."
+tags = [
+    "performance",
+    "metrics",
+    "monitoring"
+]
+categories = [
+    "Observability"
+]
++++
+
+
+How We Implemented Wallaroo's Low Overhead Performance Counters, and the Philosophy Behind Our Choices
+
+This post is based on an internal white paper from May 2016 and follows the basic paper format. The white paper's original purpose was to make a case to move away from discrete percentile measurement and use performance histogram measurements instead.
+
+## Abstract
+
+We describe two methods for computing a percentile distribution: a sorted value list, and a fixed-bin histogram approach. The two strategies are analysed for cost and their ability to answer the pertinent question of whether the target of having at least a certain fraction of the population below a specific value is met. e.g. Whether 99% of events are processed in less than 1ms, and whether 99.9% of events are processed in less than 10ms.
```

Comment:
Should there be a space between the `1` and `ms` in `1ms`? Same for `10ms`.

Reply:
I prefer it without the space. Especially later in the bin descriptions, it's easier for me to parse them as one word: `1m-10ms` rather than `1 ms - 10 ms`